### ISSUE TITLE
Update for 1.12.2. The 0.4.0 incompatible with all available optifine.

### ISSUE THREAD

**Issue by Sagedrake690** [Cadiboo/NoCubes#79]
I have tried using optifine in all pre versions and only E3 works. I found a link fo an F4 pre11 but the file is no longer there and the file is dead/not found. Im wondering if you might also be able to add in biomes o plenys list of terrain so it affects those blocks as well? it would give you a reason to backport and you can update the nether because there is a backport 1.16 mod for 1.12

**Comment by Cadiboo** [comment]
I actually tried to give this a go last week but the gradle (the tool that’s used to set up the development environment) version that 1.12.2 uses is so old that my IDE (the text-editor-on-steroids that developers use to write code) isn’t able to use it. So instead I set up 1.16 and updated the OptiFine compatibility there.
I definitely want to return to 1.12.2 sometime (it was my favourite version to mod on) but I’d like to improve the design in 1.16 and then back port it first. I’ve also forgotten how integrating with OptiFine works on 1.12.2 so I’d have to learn it again and I remember that it was very painful.
A 1.12.2 backport is definitely planned but it requires that the 1.16/1.17 code be good enough to be worth backporting, something it isn’t currently.
I’ll try to get 1.12.2 working again soon and make a release with just OptiFine compatibility fixed.
